packages/gestalt/src/BannerCallout/VRBannerCallout.tsx

```diff
@@ -0,0 +1,208 @@
+
+  const largePadding = isRtl
+    ? { paddingRight: SEMA_SPACE_800, paddingLeft: SEMA_SPACE_1200 }
+    : { paddingRight: SEMA_SPACE_1200, paddingLeft: SEMA_SPACE_800 };
+
+  const backgroundColor = MESSAGING_TYPE_ATTRIBUTES[type]?.backgroundColor;
+
+  return (
+    <Box width="100%">
+      {/*
+            SM BREAKPOINT
+      */}
+      <Box
+        borderStyle={type === 'default' ? 'sm' : undefined}
+        color={backgroundColor}
+        display="block"
+        lgDisplay="none"
+        padding={6}
+        position="relative"
+        rounding={4}
+        smDisplay="none"
+        width="100%"
+      >
+        <Flex direction="column" width="100%">
+          <Flex.Item minWidth={0}>
+            <HeaderSection
+              gap={3}
+              iconAccessibilityLabel={iconAccessibilityLabel}
+              iconSize={24}
+              marginBottom={primaryAction || secondaryAction ? 4 : undefined}
+              message={message}
+              title={title}
+              type={type}
+            />
+          </Flex.Item>
+
+          {(primaryAction || secondaryAction) && (
+            <Footer
+              buttonSize="md"
+              checkWrapped
+              marginTop={4}
+              primaryAction={primaryAction}
+              secondaryAction={secondaryAction}
+              type={type}
+            />
+          )}
+
+          {dismissButton && <DismissButton dismissButton={dismissButton} size="sm" />}
+        </Flex>
+      </Box>
+      {/*
+            MD BREAKPOINT
+      */}
+      <Box
+        borderStyle={type === 'default' ? 'sm' : undefined}
+        color={backgroundColor}
+        dangerouslySetInlineStyle={{ __style: largePadding }}
+        display="none"
+        lgDisplay="none"
+        paddingY={8}
+        position="relative"
+        rounding={4}
+        smDisplay="block"
+        width="100%"
+      >
+        <Flex direction="column" width="100%">
+          <Flex.Item minWidth={0}>
+            <HeaderSection
+              gap={6}
+              iconAccessibilityLabel={iconAccessibilityLabel}
+              iconSize={32}
+              message={message}
+              title={title}
+              type={type}
+            />
+          </Flex.Item>
+
+          {(primaryAction || secondaryAction) && (
+            <Flex.Item flex="grow">
+              <Footer
+                buttonSize="lg"
+                marginTop={6}
+                primaryAction={primaryAction}
+                secondaryAction={secondaryAction}
+                type={type}
+              />
+            </Flex.Item>
+          )}
+
+          {dismissButton && <DismissButton dismissButton={dismissButton} />}
+        </Flex>
+      </Box>
+      {/*
+            LG BREAKPOINT
+      */}
+      <Box
+        borderStyle={type === 'default' ? 'sm' : undefined}
+        color={backgroundColor}
+        dangerouslySetInlineStyle={{ __style: largePadding }}
+        display="none"
+        lgDisplay="block"
```

Comment:
Is this the only thing that's changing? This seems like a lot of dupe

Can we move into a function or something called `getResponsiveSize(size:'sm|'lg|'md')`

Reply:
I'm not sure, but also this approach may have React do 3X the work because the components would still be loaded into the virtual dom I think - since it's hidden by CSS. I think ideally this could be 3 separate smaller components, or abstracted at a different level